src/libraries/Common/tests/System/IO/Compression/CompressionStreamUnitTestBase.cs

```diff
@@ -494,6 +496,58 @@ async Task<long> GetLengthAsync(CompressionLevel compressionLevel)
             Assert.True(fastestLength >= optimalLength);
             Assert.True(optimalLength >= smallestLength);
         }
+
+        [Theory]
+        [MemberData(nameof(ZLibOptionsRoundTripTestData))]
+        public async Task RoundTripWithZLibCompressionOptions(string testFile, ZLibCompressionOptions options)
+        {
+            var uncompressedStream = await LocalMemoryStream.readAppFileAsync(testFile);
+            var compressedStream = CompressTestFile(uncompressedStream, options);
+            using var decompressor = CreateStream(compressedStream, mode: CompressionMode.Decompress);
+            var decompressorOutput = new MemoryStream();
+            int _bufferSize = 1024;
+            var bytes = new byte[_bufferSize];
+            bool finished = false;
+            int retCount;
+            while (!finished)
+            {
+                retCount = await decompressor.ReadAsync(bytes, 0, _bufferSize);
+
+                if (retCount != 0)
+                    await decompressorOutput.WriteAsync(bytes, 0, retCount);
+                else
+                    finished = true;
+            }
```

Comment:
Could this just be `await decompressor.CopyToAsync(decompressorOutput)` and then also clean up the bytes array, _bufferSize (which shouldn't have the underscore prefix anyway), etc.?